### DISCUSSION TITLE
feature request: availability of manual clock ticks on function keys

### DISCUSSION THREAD

**Discussion by primechaser** [logisim-evolution/logisim-evolution#1374] [Ideas]
Hi Team,

while using your application intensively, I thought it would be useful, 
if manual tick half and full cycle is also being made availabe on function keys next to simulation tab.

Thanks for considering

**Comment by dbh201** [comment]
Something like this?

![button_move](https://user-images.githubusercontent.com/47301809/149979371-191c31ca-fc2b-4163-9a89-d76e022c1ce7.PNG)

You can also use <kbd>Ctrl</kbd>-<kbd>T</kbd> to half-tick, and <kbd>F9</kbd> to full tick.

**Reply by primechaser** [reply]
yes.

perhaps we should consider to harmonize and complete settings
e.g.

F5: stop simulator
F6: step simulator once
F7: enable clock ticks
F8: tick clock one half cycle
F9: tick clock one full cycle

**Reply by maehne** [reply]
All these menu commands have already associated keyboard shortcuts (they are listed in th menu), which have been there as long as I am using Logisim-evolution, i.e., at least 6 years. Personally, I use mainly `Ctrl+T` (toggling a half cycle), `CTRL-K` (auto tick enable), and occasionally `Ctrl-R`(reset simulation). They cover 90 % of my needs when using the tool for demonstrations in the class room. Harmonising them to some other shortcuts should be done in a perspective to align them with shortcuts with similar semantics in other common development tools. I am thinking, e.g., of the shortcuts used to control execution of a program in a debugger / IDE. Still, people might want to customise this. So besides sane keyboard shortcut defaults, there should probably be a GUI to customise them. This is to my knowledge still lacking in Logisim-evolution. PRs in this direction are welcome!

**Reply by primechaser** [reply]
thanks for commenting. Perhaps it's an overkill at this point in time, but I agree getting these shortcuts aligned with most commmon IDEs people use would be a benefit.
I see the ability to customize settings as a second priority in this context

**Reply by maehne** [reply]
@primechaser: We are open for proposals and PRs.

**Reply by primechaser** [reply]
thanks, @maehne. As I am not into Java programming and current IDEs at all, even though aware of eclipse and so forth, I can't really come up with a standardized view based on common IDEs developers use these days.
I stated the request initially, because I didn't realize shortcuts where already existing, when I was in need of such a feature.

We might want to hear other users, more experienced than me in programming and using common tools...

**Reply by maehne** [reply]
Thanks for your reply @primechaser. I agree that we should for the moment stay with the current shortcuts. If more people raise the wish to modify/customize them, we can revisit the issue.
